From a607ede13903c22cf6dbccb86ea66da50babc8aa Mon Sep 17 00:00:00 2001
From: Kedareswara rao Appana <appana.durga.rao@xilinx.com>
Date: Wed, 23 Sep 2015 13:21:28 +0530
Subject: [PATCH 770/827] net: ethernet: xilinx: Fix issues with the optional phy_type property

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

The xlnx,phy-type is a optional DT property.
If it is not specified in the device tree then it defaults to 0
Which is the value for XAE_PHY_TYPE_MII which then causes an error
When trying to switch from 10/100 to 1000 when in SGMII mode.

Reported-by: John Linn <john.linn@xilinx.com>
Signed-off-by: Kedareswara rao Appana <appanad@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit e35cbad62bce8710bf74b91b78f4931d83e711de)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index 5d4ecb2..b2f19d2 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -2010,6 +2010,12 @@ static int axienet_probe(struct platform_device *pdev)
 	 * the device-tree and accordingly set flags.
 	 */
 	of_property_read_u32(pdev->dev.of_node, "xlnx,rxmem", &lp->rxmem);
+
+	/* The phy_type is optional but when it is not specified it should not
+	 *  be a value that alters the driver behavior so set it to an invalid
+	 *  value as the default.
+	 */
+	lp->phy_type = ~0;
 	of_property_read_u32(pdev->dev.of_node, "xlnx,phy-type", &lp->phy_type);
 	if (of_device_is_compatible(pdev->dev.of_node, "xlnx,ten-gig-eth-mac"))
 		lp->is_10Gmac = 1;
-- 
2.9.3

